### DISCUSSION TITLE
Translate constant file

### DISCUSSION THREAD

**Discussion by AndonMitev** [i18next/i18next#2094] [General]
Hello i've read other articles and resources but feel like i'am failing to do it:

So `i18n` with `useTranslation()` is working  in components all good. But now i do have constant file and i'am using it as:

```
import i18n from '../i18n';

export const sec1: SectionType = {
  title: i18n.t('test.text1'),
};
```

also tried with: `import i18n from "i18next"` but still not working, this my i18n config:

```
import i18n from 'i18next';
import LanguageDetector from 'i18next-browser-languagedetector';
import Backend from 'i18next-http-backend';
import { initReactI18next } from 'react-i18next';

i18n
  .use(Backend)
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    fallbackLng: 'ja',
    debug: true,
    interpolation: {
      escapeValue: false,
    },
  });

export default i18n;

```

any ideas if i'm missing something or so?

PS: If i update constant to function like this:

```
import i18n from '../i18n';

export const sec1 = () => ({
  title: i18n.t('test.text1'),
});

```

is working tho, but is it possible to make it work without converting constant to a function

**Comment by jamuhl** [comment]
https://github.com/i18next/react-i18next/issues/1698#issuecomment-1822433433
